client/src/context/AuthContext.js

```diff
@@ -0,0 +1,101 @@
+import React, { useReducer } from 'react';
+import axios from 'axios';
+
+
+
+export const AuthContext = React.createContext();
+
+
+const types = {
+    Set_Is_Loading: "Set_Is_Loading",
+	Set_Logged_In: "Set_Logged_In",
+    Set_User: "Set_User",
+    Set_Isnot_Graduate:'Set_Isnot_Graduate',
+    Set_Error: "Set_Error",
+    Set_Logout:"Set_Logout"
+};
+
+
+const authReducer = (state, action) => {
+	switch (action.type) {
+	case types.Set_Is_Loading:
+		return { ...state, isLoading: action.payload };
+	case types.Set_Error:
+		return { ...state, isLoading: false, error:action.payload };
+	case types.Set_Logged_In:
+		return { ...state, isAuthenticated:true, isLoading: false };
+    case types.Set_User:
+        return { ...state, userName: action.payload, isAuthenticated:true, isLoading: false };
```

Comment:
Try to keep your actions matching with your state names. I'd call this `Set_UserName`.